.github/workflows/beam_PostCommit_Python.yml

```diff
@@ -54,14 +54,15 @@ env:
 jobs:
   beam_PostCommit_Python:
     name: ${{ matrix.job_name }} (${{ matrix.job_phrase }} ${{ matrix.python_version }})
-    runs-on: [self-hosted, ubuntu-20.04, highmem22]
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 240
     strategy:
       fail-fast: false
       matrix:
         job_name: [beam_PostCommit_Python]
         job_phrase: [Run Python PostCommit]
         python_version: ['3.9', '3.10', '3.11', '3.12']
+        os: [[self-hosted, ubuntu-20.04, highmem22], [ubuntu-latest]]
```

Comment:
Could you add a small comment that describes why we're doing this here and in the other workflow?